File: Python/334A-candybags.py
n = int(input())
midpoint = n**2//2
numbers = []
for i in range(1, midpoint+1):
    numbers.append(i)
    numbers.append(n**2+1-i)
children = [numbers[i:i+n] for i in range(0, n**2, n)]
for child in children:
    for x in child:
        print(x, end=' ')
    print()

File: C++/A/334A-candybags.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, midpoint;
    cin >> n;
    midpoint = pow(n, 2) / 2;
    vector<int> numbers;
    for (int i = 1; i <= midpoint; i++)
        numbers.push_back(i), numbers.push_back(pow(n, 2) + 1 - i);
    vector<vector<int>> children;
    for (int i = 0; i < n * n; i += n)
    {
        vector<int> group;
        for (int j = i; j < i + n; j++)
            group.push_back(numbers[j]);
        children.push_back(group);
    }
    for (vector<int> child : children)
    {
        for (int x : child)
            cout << x << ' ';
        cout << '\n';
    }
}